### ISSUE TITLE
[Bug]: Using LSP Manager from #1707

### ISSUE THREAD

**Issue by ssbanerje** [LunarVim/LunarVim#1757]
### Problem description

I am trying to use the user config feature from #1707 like so:

```lua
local custom_lsp_configs = {
  dockerls = {
    root_dir = function(fname)
      return require("lspconfig").util.root_pattern(".git")(fname) or require("lspconfig").util.path.dirname(fname)
    end,
    filetypes = { "Dockerfile*", "dockerfile*" }
  },
  -- ...
  -- ...
}

local lsp_manager = require("lvim.lsp.manager")
for server, config in pairs(custom_lsp_configs) do
  lsp_manager.setup(server, config)
end
```

But this seems to be always overridden by the config that `lvim.lsp.templates_dir`. Maybe I am missing something here but how is this supposed to work?

Also, I am not sure how #1707 fixed #1697. Based on the proposal in #1697, wouldn't it make sense to have a global table (say a `lvim.lsp.user_config.LANGX`) which can actually be changed by the user in `config.lua`, and is read and merged into the actual LSP server config? 

### LunarVim version

rolling @ 9e18c16

### Neovim version (>= 0.5)

0.5.1

### I have

- [X] Read the readme
- [X] Read the wiki
- [X] Searched for similar issues

**Comment by kylo252** [comment]
> But this seems to be always overridden by the files in `lvim.lsp.templates_dir`. Maybe I am missing something here but how is this supposed to work?

There's no mechanism to override the setting in `lvim.lsp.templates_dir` because they're simply meant as a template. See https://www.lunarvim.org/languages/#overriding-the-default-configuration

**Comment by kylo252** [comment]
Sorry about the confusion. 

I left a note about implementing this. I left a note regarding this since I didn't know what is a good API for it: https://github.com/LunarVim/LunarVim/blob/9e18c16b2b378792c689591190774786fd5da04c/lua/lvim/lsp/templates.lua#L20-L23

**Comment by kylo252** [comment]
> So eventually there will be a override list and an ignored one?

I think it will just be more confusing like this. So keep lvim.lsp.override and use it for this purpose, since it's basically the same outcome. If anything it might be more annoying to leave it where it is.

Move `is_overridden` to the template generator to replace `is_ignored`, and remove the check from the lsp manager. Also populate `lvim.lsp.override` with the values that we already have in `is_ignored`. I think that should be good.

**Comment by ssbanerje** [comment]
Yes that would fix it. But the user will have to roll out their own auto/lazy-loading logic if I understand this correctly. 

Why not just reuse the template files which do the autoloading, while allowing the user to modify the config.

Continued the discussion in #1762.
